staging/src/k8s.io/legacy-cloud-providers/openstack/openstack_instances.go

```diff
@@ -44,10 +44,36 @@ const (
 	instanceShutoff = "SHUTOFF"
 )
 
+func (os *OpenStack) getCompute() *gophercloud.ServiceClient {
+	err := os.ensureCloudProviderWasInitialized()
+	if err != nil {
+		klog.Errorf("unable to access compute v2 API : %v", err)
+		return nil
+	}
+
+	compute, err := os.NewComputeV2()
+	if err != nil {
+		klog.Errorf("unable to access compute v2 API : %v", err)
+		return nil
+	}
+
+	return compute
+}
+
 // Instances returns an implementation of Instances for OpenStack.
 func (os *OpenStack) Instances() (cloudprovider.Instances, bool) {
 	klog.V(4).Info("openstack.Instances() called")
 
+	err := os.ensureCloudProviderWasInitialized()
+	if err != nil {
+		// cannot initialize cloud provider - return empty instances without compute instance,
+		// it will be generated later with any call of an Instances' receiver.
+		klog.Errorf("cannot initialize cloud provider, only limited functionality is available : %v", err)
```

Comment:
how often is this func called?

Reply:
Instances? I don't know but quite often I think. Why do you ask?